src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -129,36 +145,47 @@
-       :on-done (fn [{:keys [success? get-parsed-response status-text]}]
-                  (if success?
-                    (swap! state assoc :methods (->> (get-parsed-response)
-                                                     (map #(select-keys % [:namespace :name :snapshotId]))
-                                                     (group-by (juxt :namespace :name))
-                                                     (utils/map-values (partial map :snapshotId))))
-                    ;; FIXME: :error-message is unused
-                    (swap! state assoc :error-message status-text)))}))
+   (fn [{:keys [this]}]
+     (this :-load-validated-method-config))
+   :component-did-update
+   (fn [{:keys [state]}]
+     (when (and (not (:methods @state)) (:loaded-config @state))
+       (let [loaded-config (:loaded-config @state)
+             methodConfiguration (:methodConfiguration loaded-config)
+             methodRepoMethod (:methodRepoMethod methodConfiguration)
+             methodName (:methodName methodRepoMethod)
+             methodNamespace (:methodNamespace methodRepoMethod)]
+         (endpoints/call-ajax-orch
+          {:endpoint (endpoints/list-method-snapshots methodNamespace methodName)
+           :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                      (let [response (get-parsed-response)]
+                        (if success?
+                          (swap! state assoc :methods-response response :methods (->> response
+                                                                                      (group-by (juxt :namespace :name))
+                                                                                      (utils/map-values (partial map :snapshotId))))
+                          ;; FIXME: :error-message is unused
+                          (swap! state assoc :error-message status-text))))}))))
    :-render-display
    (fn [{:keys [props state locals this]}]
-     (let [locked? (get-in props [:workspace :workspace :isLocked])]
+     (let [locked? (get-in props [:workspace :workspace :isLocked])
+           methods (:methods @state)
+           methodRepoMethod (get-in @state [:loaded-config :methodConfiguration :methodRepoMethod])]
        [:div {}
         [comps/Blocker {:banner (:blocker @state)}]
         [mc-sync/SyncContainer (select-keys props [:workspace-id :config-id])]
         [:div {:style {:padding "1em 2em" :display "flex"}}
          [Sidebar (merge (select-keys props [:access-level :workspace-id :after-delete])
-                         (select-keys @state [:editing? :loaded-config])
+                         (select-keys @state [:editing? :loaded-config :redacted?])
                          (select-keys @locals [:body-id])
-                         {:parent this :locked? locked?})]
+                         {:parent this :locked? locked? :snapshots (get methods [(:methodNamespace methodRepoMethod) (:methodName methodRepoMethod)])})]
```

Comment:
@MattPutnam is there a cleaner way to do this?

Reply:
```clojure
[(:methodNamespace methodRepoMethod) (:methodName methodRepoMethod)]
;; equals
(replace methodRepoMethod [:methodNamespace :methodName])
```